src/Event.cc

```diff
@@ -375,7 +447,70 @@ PropertySet::Ptr Event::processTextMessage(cms::TextMessage* textMessage) {
 
     return unmarshall(text);
 }
+
+/** private method to try to add data to a property set
+ * \param typeInfo a string containing the name of the data type
+ * \param item a node of a boost::property_tree::ptree containing data
+ * \param key the name of the data
+ * \param ps a PropertySet to store the name and data into.
+ * \return true if data was added to the PropertySet
+  */
+bool Event::addDataItem(std::string typeInfo, boost::property_tree::ptree& item, std::string key, PropertySet& ps) {
+    if (typeInfo == "string") {
+        std::string value = item.get_value<std::string>();
+        ps.add(key, value);
+    } else if (typeInfo == "bool") {
+        bool value = item.get_value<bool>();
+        ps.add(key, value);
+    } else if (typeInfo == "long") {
+        long value = item.get_value<long>();
+        ps.add(key, value);
+    } else if (typeInfo == "long long") {
+        long long value = item.get_value<long long>();
+        ps.add(key, value);
+    } else if (typeInfo == "int") {
+        int value = item.get_value<int>();
+        ps.add(key, value);
+    } else if (typeInfo == "float") {
+        float value = item.get_value<float>();
+        ps.add(key, value);
+    } else if (typeInfo == "double") {
+        double value = item.get_value<double>();
+        ps.add(key, value);
+    } else if (typeInfo == "datetime") {
+        long long value = item.get_value<long long>();
+        ps.add(key, lsst::daf::base::DateTime(value, lsst::daf::base::DateTime::UTC));
+    } else {
+        return false;
+    }
+    return true;
+}
+
+/** private method to iterate through the representation of a propertyset
+  * within the boost::property_tree::ptree
+  * \param child a boost::property_tree::ptree to iterate through
+  * \return a PropertySet::Ptr containing the data that was stored in child.
+  */
+PropertySet::Ptr Event::parsePropertySet(boost::property_tree::ptree child) {
+    PropertySet::Ptr psp(new PropertySet);
+    
+    BOOST_FOREACH(boost::property_tree::ptree::value_type const &v, child.get_child("")) {
```

Comment:
Is use of `BOOST_FOREACH` permitted?  Is this something you can't do with regular iterators?